amy/dashboard/views.py

```diff
@@ -238,25 +269,33 @@ def training_progress(request):
         get_involved_form = GetInvolvedForm(
             data=request.POST, instance=base_training_progress
         )
+
         if get_involved_form.is_valid():
-            get_involved_form.save()
+            return self.form_valid(get_involved_form)
 
-            messages.success(
-                request,
-                "Thank you. Your Get Involved submission will be reviewed within 7-10 "
-                "days.",
-            )
-            return redirect(reverse("training-progress"))
+        else:
+            return self.form_invalid(get_involved_form)
```

Comment:
This logic should not be handled in `post()` method. If you need to set instance, you can probably override `get_form_kwargs()`:

```python
    def get_form_kwargs(self):
        kwargs = super().get_form_kwargs()
        base_training_progress = TrainingProgress(
            trainee=self.request.user,
            state="n",  # not evaluated yet
            requirement=TrainingRequirement.objects.get(name="Get Involved"),
        )
        kwargs["instance"] = base_training_progress
        return kwargs
```

There probably is no access to `request` anymore and so it should be saved in `dispatch()` or `get()` and `post()` methods. We have a few views that use this technique.

Reply:
Ah, that's much better, thank you